### PR TITLE
CoW: Avoid copying Index in Series constructor

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/indexes/interval.py [hunk 1]
```diff
@@ -399,7 +399,8 @@ def inferred_type(self) -> str:
         """Return a string of the type inferred from the values"""
         return "interval"
 
-    @Appender(Index.memory_usage.__doc__)
+    # Cannot determine type of "memory_usage"
+    @Appender(Index.memory_usage.__doc__)  # type: ignore[has-type]
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/interval.py:403]: How is this related?

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

File: pandas/core/indexes/interval.py (hunk 1 above)
Comment (phofl) [reply pandas/core/indexes/interval.py:403]: This seems like a mypy bug. 

Adding 
```
from pandas._libs.internals import BlockValuesRefs
```

to the base file causes this
